File: client/bcephal-client-model/Grids/Filters/ColumnFilter.cs
using Bcephal.Models.Filters;
using System;
using System.Collections.Generic;
using System.Text;

namespace Bcephal.Models.Grids.Filters
{
    public class ColumnFilter
    {

        public string Name { get; set; }

        //public GrilleColumn Column { get; set; }
        public DimensionType dimensionType { get; set; }

        public long? dimensionId { get; set; }

        public string Link { get; set; }

        public string Value { get; set; }

        public string Operation { get; set; }

        public List<ColumnFilter> Items { get; set; }

        public bool Grouped { get; set; }

        public long? JoinColumnId;

        public ColumnFilter()
        {
            this.Items = new List<ColumnFilter>(0);
            this.Grouped = false;
        }

    }
}

File: server/bcephal-base/src/main/java/com/moriset/bcephal/domain/filters/ColumnFilter.java
/**
 * 
 */
package com.moriset.bcephal.domain.filters;

import java.util.ArrayList;
import java.util.List;

import org.springframework.util.StringUtils;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.dimension.DimensionType;

import lombok.Data;

/**
 * @author Joseph Wambo
 *
 */
@Data
public class ColumnFilter {

	private String name;

	private String link;

	private String value;

	private String operation;

	private List<ColumnFilter> items;

	private boolean grouped;

	private DimensionType dimensionType;

	private Long dimensionId;
	
	private Long joinColumnId;
	//for server filter
	@JsonIgnore
	private boolean isJoin;
	//for server filter
	@JsonIgnore
	private String joinName;
	
	@JsonIgnore
	private Class<?> Type;

	public ColumnFilter() {
		this.items = new ArrayList<ColumnFilter>(0);
		this.grouped = false;
		this.isJoin = false;
	}
	
	@JsonIgnore
	public boolean isSortFilter() {
		return StringUtils.hasText(getOperation()) &&  getOperation().equals(BrowserDataFilter.SortBy);
	}
}
